mlos_bench/mlos_bench/config/schemas/services/remote/azure/azure-auth-service-subschema.json

```diff
@@ -23,6 +23,18 @@
                         "tokenRequestInterval": {
                             "description": "Time before token expiration to request a new token",
                             "type": "number"
+                        },
+                        "keyVaultName": {
+                            "description": "Name of key vault to retrieve the certificate from, for authorizing the Service Principal",
+                            "type": "string"
+                        },
+                        "certName": {
+                            "description": "Name of the Service Principal's certificate in the key vault",
+                            "type": "string"
+                        },
+                        "spClientId": {
```

Comment:
Can we have a config file with these keys at `config/services/remote/azure/service-auth.jsonc`?
Maybe use the placeholder values like in `config/services/remote/azure/service-fileshare.jsonc`

Reply:
Initially I added these keys into that file with placeholder values.

However because we end up trying to use the keyvault when those keys are present, the unit tests around loading environments end up failing because those interactions are not mocked out.
I will try to see if I can modify those unit tests!

Reply:
I moved the initialization for the service principal into a separate initialization function to avoid keyvault-specific patching for those general environment loading tests.